internal/pkg/service/common/etcdop/watch.go

```diff
@@ -75,56 +79,52 @@ func (v Prefix) GetAllAndWatch(ctx context.Context, client *etcd.Client, handleE
 		if err != nil {
 			// GetAll error is fatal
 			handleErr(err)
-			close(ch)
+			close(outCh)
+			close(initDoneCh)
+			return
 		}
 
 		// Continue with Watch where GetAll ended
+		close(initDoneCh)
 		opts = append(opts, etcd.WithRev(itr.Header().Revision+1))
-		v.doWatch(ctx, client, handleErr, ch, opts...)
+		v.doWatch(ctx, client, handleErr, outCh, opts...)
 	}()
 
-	return ch
+	return outCh, initDoneCh
 }
 
-func (v Prefix) doWatch(ctx context.Context, client etcd.Watcher, handleErr func(err error), ch chan Event, opts ...etcd.OpOption) {
+func (v Prefix) doWatch(ctx context.Context, client etcd.Watcher, handleErr func(err error), outCh chan Event, opts ...etcd.OpOption) {
 	opts = append([]etcd.OpOption{etcd.WithPrefix()}, opts...)
-	rawCh := client.Watch(ctx, v.Prefix(), opts...)
 	go func() {
+		defer close(outCh)
+
+		// In case of an error, the watch channel can be closed.
+		// It will be recreated, and it will continue from the last revision.
+		revision := int64(0)
```

Comment:
Podla [dokumentacie](https://github.com/etcd-io/etcd/blob/16e1fff519eeff66e626dd15fef399ea2b10b9cc/client/v3/watch.go#L58-L85) sa môže stať, že v pripade chyby sa `rawCh` z etcd uzavrie.

Ak by sa to stalo, tak chceme `rawCh` znova vytvorit a pokracovat od posledneho spracovaneho eventu.
Skonci chceme iba ked `ctx.Done`.

Reply:
Aby som to nemusel riesit na inych urovniach, tak som to upravil rovno tu, na tej najnizsej urovni.